docs/specification.md

```diff
@@ -1,150 +1,371 @@
+
+---
+### 3. Ancillary attribute management: recommended non-inherent attributes
 
-#### Algorithm:
+In *Section 1: Encoding*, we distinguished between *inherent* and *non-inherent* attributes. Non-inherent attributes provide a standardized way for implementations to store and serve additional, third-party attributes that do not contribute to digest. As long as separate implementations keep such information in non-inherent attributes, the identifiers will remain compatibile. Furthermore, the structure for how such non-inherent metadata is retrieved will be standardized. Here, we specify standardized, useful non-inherent attributes that we recommend.
 
+#### 3.2 The `sorted_name_length_pairs` attribute (`RECOMMENDED`)
 
-**Step 1**. Repeat for each sequence in the collection:  
+The `sorted_name_length_pairs` attribute is a *non-inherent* attribute of a sequence collection with a formal definition, provided here. It is `RECOMMENDED` that all seqcol implementations add this attribute to all sequence collections. When digested, this attribute provides an identifier for an order-invariant coordinate system for a sequence collection. Because it is *non-inherent*, it does not affect the identity (digest) of the collection. It is created deterministically from the `names` and `lengths` attributes in the collection; it *does not* depend on the actual sequence content, so it is consistent across two collections with different sequence content if they have the same `names` and `lengths`, which are correctly collated, but with pairs not necessarily in the same order. For rationale and use cases of `sorted_name_length_pairs`, see *Footnote F6*.
 
-- 1a. Compute the refget digest for the sequence.
-- 1b. Compute the length of the sequence.
-- 1c. Concatenate the name, length, and refget digest, in the order provided in the collection, delimited by `ATTRIBUTE_DELIMITER`.
-- 1d. Compute the digest of the result of step 1c. This is the *annotated sequence digest* for the sequence.
+Algorithm: 
 
-**Step 2**. Concatenate the computed annotated sequence digests, in the order provided in the collection, delimited by `ITEM_DELIMITER`.
+1. Lump together each name-length pair from the primary collated `names` and `lengths` into an object, like `{"length":123,"name":"chr1"}`.
+2. Canonicalize JSON according to the seqcol spec (using RFC-8785).
+3. Digest each name-length pair string individually.
+4. Sort the digests lexographically.
+5. Add as a non-inherent, non-collated attribute to the sequence collection object.
 
-**Step 3**. Compute the digest of the result of step 2. This is the *seqcol digest*, the final unique identifier for this sequence collection.
+## Footnotes
```

Comment:
Many of the footnotes are linked to elements in the ADR. I think we should be linking out to the relevant ADRs in addition to this higher level description which might summarise one or more ADRs

Reply:
done!